File: src/main/java/com/nickcoblentz/montoya/settings/IExtensionSetting.java
package com.nickcoblentz.montoya.settings;

public interface IExtensionSetting {
    String getCurrentValue();

    void setCurrentValue(String currentValue);

    void saveToProject();

    void loadFromProject();

    void saveToPreferences();

    void loadFromPreferences();

    void save();
    void load();

    String getName();

    String getStorageKey();

    String reset();
}

File: src/main/java/com/nickcoblentz/montoya/settings/IExtensionSetting.kt
package com.nickcoblentz.montoya.settings

interface IExtensionSetting {
    var currentValue : String
    val defaultSaveLocation: ExtensionSettingSaveLocation
    fun saveToProject()

    fun loadFromProject()

    fun saveToPreferences()

    fun loadFromPreferences()

    fun save()
    fun load()

    val name: String

    val storageKey: String

    fun reset(): String?
}